app/services/broadcasts/welcome_notification/generator.rb

```diff
@@ -11,19 +11,21 @@ def self.call(*args)
       end
 
       def call
-        return if send_welcome_notification || commented_on_welcome_thread?
+        return if commented_on_welcome_thread? || send_welcome_notification?
 
         welcome_broadcast = Broadcast.find_by(title: "Welcome Notification: welcome_thread")
         Notification.send_welcome_notification(receiver_id, welcome_broadcast.id)
       end
 
       def send_welcome_notification?
-        # welcome_broadcast = Broadcast.where(user_id: receiver_id)
-        # Notification.where(notifiable_id: welcome_broadcast.id, user_id: receiver_id).any?
         user = User.find_by(id: receiver_id)
-        user.notifications.include?(welcome_broadcast)
-      rescue ArgumentError, NoMethodError => e
-        Rails.logger.error("#{user_id} is not a valid id: #{e}")
+        welcome_broadcast = Broadcast.find_by(active: true)
+        Notification.where(notifiable_id: welcome_broadcast.id, user_id: receiver_id) && user.notifications.include?(welcome_broadcast)
```

Comment:
A few issues here:
1. Why would a user's notifications include a broadcast (`user.notifications.include?(welcome_broadcast)`)? That seems incorrect to me. A user's notifications should only include notifications, right? Not any other object?
2. This line is actually just not being used at all. If you want to do this check, then you need some kind of conditional here? Right now, you are checking if a specific notification exists, and whether the user's notifications include the `welcome_broadcast`, but you're not _doing_ anything with this line...and then you trigger the Rails logger with an error here by default. I think you probably need to rework this to take these checks into consideration. 
3. You will only want to call the rails logger with an error if there actually is an error. Right now, it's always being called. What is the error that we care about and that we care about sending to the rails logger? Is it an `ArgumentError` or a `RecordInvalid` error? Whenever that error happens, we want to `rescue` from it, and then log to the rails logger.

Reply:
Answers to your questions:

1. Yes, you are correct. I think to check whether the Notification already exists for that User, I would have to do something like: `if receiving_user.notifications.notifiable == welcome_broadcast`...
2. A conditional here makes sense, yes! And now it makes sense as to why the `logger` is being triggered and my spec is failing. 🙃 
3. I have the `rescue` commented out below, but I have since commented it back in so that as I rework this file, the `logger` will hopefully work as expected.